internal/cmd/controller/grutil/status.go

```diff
@@ -99,6 +107,64 @@ func SetStatusFromBundleDeployments(ctx context.Context, c client.Client, gitrep
 	return nil
 }
 
+// SetStatusFromGitjob sets the status fields relative to the given job in the gitRepo
+func SetStatusFromGitjob(ctx context.Context, c client.Client, gitRepo *fleet.GitRepo, job *batchv1.Job) error {
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(job)
+	if err != nil {
+		return err
+	}
+	uJob := &unstructured.Unstructured{Object: obj}
+
+	result, err := status.Compute(uJob)
+	if err != nil {
+		return err
+	}
+
+	terminationMessage := ""
+	if result.Status == status.FailedStatus {
+		selector := labels.SelectorFromSet(labels.Set{"job-name": job.Name})
+		podList := &corev1.PodList{}
+		err := c.List(ctx, podList, &client.ListOptions{LabelSelector: selector})
+		if err != nil {
+			return err
+		}
+
+		sort.Slice(podList.Items, func(i, j int) bool {
+			return podList.Items[i].CreationTimestamp.Before(&podList.Items[j].CreationTimestamp)
+		})
+
+		terminationMessage = result.Message
+		if len(podList.Items) > 0 {
+			for _, podStatus := range podList.Items[len(podList.Items)-1].Status.ContainerStatuses {
+				if podStatus.Name != "step-git-source" && podStatus.State.Terminated != nil {
+					terminationMessage += podStatus.State.Terminated.Message
+				}
+			}
+		}
+	}
+
+	gitRepo.Status.GitJobStatus = result.Status.String()
+	gitRepo.Status.ObservedGeneration = gitRepo.Generation
```

Comment:
Do we deliberately not set the `UpdateGeneration` here? If so, why?

Reply:
I moved that to the place where it is checked:

```go
func (r *GitJobReconciler) deleteJobIfNeeded(ctx context.Context, gitRepo *v1alpha1.GitRepo, job *batchv1.Job) error {
	logger := log.FromContext(ctx)
	// if force delete is set, delete the job to make sure a new job is created
	if gitRepo.Spec.ForceSyncGeneration != gitRepo.Status.UpdateGeneration {
		gitRepo.Status.UpdateGeneration = gitRepo.Spec.ForceSyncGeneration
		logger.Info("job deletion triggered because of ForceUpdateGeneration")
		if err := r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil && !errors.IsNotFound(err) {
			return err
		}
	}
```

I think it makes more sense to set the value where it is checked so the logic is at the same scope. 

In fact, I did the same for `gitRepo.Status.ObservedGeneration = gitRepo.Generation`
I moved it to the place where it is checked also.

This way the `SetStatusFromGitjob` function sets only fields related to the Gitjob.